MetaMask/metamask-mobile
# Add browser cookies support on Android

**Description**

At the moment users are not able to use cookies on Android since the HTTP client (OkHttp) is [rejecting cookies by default](https://square.github.io/okhttp/4.x/okhttp/okhttp3/-cookie-jar/), this PR implements an interface that saves and loads cookies accordingly.

This PR also fixes clearing cookies on iOS and improves the "Clear Cookies" section structure.

**Test Cases**

Case 1 (Android):
 - Access a website that has cookies or login into one that has session cookies (ex.: Go to https://notareal.club/shop)
 - Add a product to your cart.
- Click view cart or navigate to the cart page.
- You should see an item in the cart

Cases 2 (iOS):
 - Access a website that has cookies or login into one that has session cookies (ex.: Go to https://notareal.club/shop)
 - Go to Settings -> Security & Privacy -> Clear Browser Cookies
 - Clear cookies button should be disabled
 - Navigate to the browser you should have no cookies

**Checklist**


**Screenshots/Recordings**


https://user-images.githubusercontent.com/575479/178604012-9db220e4-f3a1-4aca-8f98-f2a10269f9a3.mp4


https://user-images.githubusercontent.com/575479/178604025-0a7f9f5d-cc4d-4a1f-bf00-5ba84a17dae2.mp4


https://user-images.githubusercontent.com/575479/178604034-9f017d65-fa62-450d-b80f-d7318dcf36e8.mp4



**Issue**

Progresses #4245, #2954, #2544, #4190
